streaming/base/storage/upload.py

```diff
@@ -478,6 +562,39 @@ def check_bucket_exists(self, remote: str):
                               f'Check the bucket permission or create the bucket.',)
             raise error
 
+    def list_objects(self, prefix: Optional[str] = None) -> Optional[List[str]]:
+        """List all objects in the OCI object store with the given prefix.
+
+        Args:
+            prefix (Optional[str], optional): The prefix to search for. Defaults to ``None``.
+
+        Returns:
+            List[str]: A list of object names that match the prefix.
+        """
+        if prefix is None:
+            prefix = ''
+
+        obj = urllib.parse.urlparse(self.remote)
+        bucket_name = obj.netloc.split('@' + self.namespace)[0]
+        prefix = os.path.join(str(obj.path).strip('/'), prefix)
+
+        object_names = []
+        next_start_with = None
+        response_complete = False
+        try:
+            while not response_complete:
+                response = self.client.list_objects(namespace_name=self.namespace,
+                                                    bucket_name=bucket_name,
+                                                    prefix=prefix,
+                                                    start=next_start_with).data
+                object_names.extend([resp_obj.name for resp_obj in response.objects])
+                next_start_with = response.next_start_with
+                if not next_start_with:
+                    response_complete = True
+            return object_names
+        except Exception as _:
```

Comment:
This is too generic. Is there a specific Exception that `list_objects` returns when the file is not found? Or something like KeyError?

Reply:
Added two specific errors.